server/game/cards/01_SOR/units/BlackOneScourgeOfStarkillerBase.ts

```diff
@@ -11,13 +11,13 @@ export default class BlackOneScourgeOfStarkillerBase extends NonLeaderUnitCard {
 
     public override setupCardAbilities() {
         this.addTriggeredAbility({
-            title: 'You may discard your hand',
+            title: 'Discard your hand',
             when: {
                 onCardPlayed: (event, context) => event.card === context.source,
                 onCardDefeated: (event, context) => event.card === context.source,
             },
             optional: true,
-            immediateEffect: AbilityHelper.immediateEffects.discardHand(),
+            immediateEffect: AbilityHelper.immediateEffects.discardEntireHand((context) => ({ target: context.player })),
```

Comment:
I have to confess I'm not 100% sure how `context.player` is set in all situations so we typically use an explicit reference to the host card's controller via `context.source.controller`